src/components/CommandTable.tsx

```diff
@@ -116,108 +118,141 @@ if (commandsUngrouped.length > 0) {
+            alignItems: 'center',
+            pointerEvents: 'none',
+          })}
+        >
+          <SearchIcon size={16} fill={isLightTheme ? '#666' : '#999'} />
+        </div>
+        <input
+          type='text'
+          placeholder='Search gestures...'
+          onInput={(e: React.ChangeEvent<HTMLInputElement>) => {
+            onInput?.(e.target.value)
+          }}
+          className={css({
+            marginLeft: 0,
+            marginBottom: 0,
+            boxSizing: 'border-box',
+            width: '100%',
+            minWidth: '100%',
+            paddingLeft: '2rem',
+            borderRadius: '8px',
+          })}
+        />
+      </div>
     </div>
   )
 }
 
-/** Renders a table of commands. */
+/** Renders a table of commands with a fade-in animation when sorting changes. */
 const CommandTable = ({
   customize,
   onSelect,
-  selectedCommand: selectedCommand,
+  selectedCommand,
 }: {
   customize?: boolean
   onSelect?: (command: Command | null) => void
   selectedCommand?: Command
 }) => {
   const [search, setSearch] = useState('')
   const commands = useFilteredCommands(search, { platformCommandsOnly: true })
+  const [sortOrder, setSortOrder] = useState<'alphabetical' | 'type'>('type')
+  const [previousSortOrder, setPreviousSortOrder] = useState(sortOrder)
+  const [isFading, setIsFading] = useState(false)
+
+  useEffect(() => {
+    if (sortOrder !== previousSortOrder) {
+      setIsFading(true) // Start fading out
+      setTimeout(() => {
+        setPreviousSortOrder(sortOrder) // Update to new view after fade out
+        setIsFading(false) // Start fading in
+      }, 250) // Matches transition duration
+    }
+  }, [sortOrder, previousSortOrder])
 
   return (
     <div>
-      <SearchCommands onInput={setSearch} />
-      <div className={css({ textAlign: 'left' })}>
-        {search ? (
-          <CommandsGroup
-            title={'Results'}
-            commands={commands}
-            selectedCommand={selectedCommand}
-            customize={customize}
-            onSelect={onSelect}
-            search={search}
-          />
-        ) : (
-          groups.map(group => {
-            const commands = group.commands
-              .map(commandById)
-              .filter((command): command is Command => (isTouch ? !!command.gesture : !!command.keyboard))
+      <div
+        className={css({
+          display: 'flex',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          gap: '5px',
+        })}
+      >
+        <SearchCommands onInput={setSearch} />
+        <SortButton onSortChange={setSortOrder} />
+      </div>
 
-            // do not render groups with no commands on this platform
-            return commands.length > 0 ? (
+      {/* Smooth Fade-in Transition */}
+      <div
+        className={css({
+          textAlign: 'left',
+          opacity: isFading ? 0 : 1,
+          transition: 'opacity 0.3s ease-in-out',
+        })}
+      >
```

Comment:
Would it be possible to use the existing [FadeTransition](https://github.com/cybersemics/em/blob/37f3e98916904810830c18b92436cad2016fac47/src/components/FadeTransition.tsx) component? This helps abstract away the timing and animation details.

Note that all of our animation durations are defined in https://github.com/cybersemics/em/blob/37f3e98916904810830c18b92436cad2016fac47/src/durations.config.ts for consistency.